### PR TITLE
[android] Make set_up_android.py test SDK path always

### PR DESCRIPTION
**Pull request by pastk**
Make the script always check the given path contains Android SDK (there was no check for command line option before).
Made it more descriptive/verbose and cleaned up unused code also.

### CODE REVIEW

**Comment by pastk** [comment]
@rtsisyk could you please review? As it seems you were the last one to modify the script.

File: tools/android/set_up_android.py [hunk 1]
```diff
@@ -37,59 +37,61 @@ def read_local_properties():
           sdkDir = line.split('=')[1].strip()
   return sdkDir
 
+def test_path(title, path, subpath):
+  test = os.path.join(path, *subpath)
```
Comment (biodranik) [inline tools/android/set_up_android.py:41]: We plan to migrate all scripts to python3. Does it make sense to use Path here? Generally, the code can be more readable and take less lines :)

File: tools/android/set_up_android.py [hunk 2]
```diff
@@ -37,59 +37,61 @@ def read_local_properties():
...
+  if path and os.path.isfile(test):
+    return os.path.abspath(path)
+  else:
```
Comment (biodranik) [inline tools/android/set_up_android.py:44]: Else is redundant.

**Review by biodranik — APPROVED** [review]
Do we need this script at all? Is there any way to build without using it?

**Comment by pastk** [comment]
> Do we need this script at all? Is there any way to build without using it?

I don't really know, I'm new to the project and to android dev in particular.
My guess its needed only for command line builds so that gradle knows where is the SDK. But according to the docs on Win platform it works out of the box...

File: tools/android/set_up_android.py (hunk 1 above)
Comment (pastk) [reply tools/android/set_up_android.py:41]: Sorry, I didn't get it..
(and I have a small experience with python so far)
How is it related to migration to 3?
Do you suggest passing preconcatenated path+subpath to the function?
Could you be more specific please?

Comment (biodranik) [reply tools/android/set_up_android.py:41]: https://docs.python.org/3/library/pathlib.html

**Comment by biodranik** [comment]
> My guess its needed only for command line builds so that gradle knows where is the SDK. But according to the docs on Win platform it works out of the box...

Does command line build in Linux/Mac work without local.properties ?

**Comment by pastk** [comment]
> > My guess its needed only for command line builds so that gradle knows where is the SDK. But according to the docs on Win platform it works out of the box...
> 
> Does command line build in Linux/Mac work without local.properties ?

I was wrong. Even Android Studio fail without that setting..
```
A problem occurred evaluating root project 'android'.
SDK location not found. Define location with an ANDROID_SDK_ROOT environment variable or by setting the sdk.dir path in your project's local properties file at '/home/pastk/dev/organicmaps/android/local.properties'.
```
Makes sense as Android Studio launches a stand-alone gradle daemon by default.
I think for new users running a script is easier than setting (and persisting) an env var so its better to keep the script.

(I don't know why it works (and is it?) out-of-the-box in Win; though there are many Windows guides online how to put sdk location into local.properties...)

File: tools/android/set_up_android.py (hunk 1 above)
Comment (pastk) [reply tools/android/set_up_android.py:41]: > For low-level path manipulation on strings, you can also use the os.path module. 

Yeap, pathlib is more modern, but [`os.path`](https://docs.python.org/3/library/os.path.html) is not deprecated either.

It'll be good to have a coordinated migration effort.
But I would rather separate migration from this PR's scope, unless you insist on it.

**Review by biodranik — APPROVED** [review]

**Comment by biodranik** [comment]
If we touch some old code, we usually improve it, because there will be no more time to come back and do improvements in the second iteration :)

**Pull request merged**